### PR TITLE
Remove/dedupe/upgrade several dependencies

### PR DESCRIPTION
**Pull request by j-f1**

### CODE REVIEW

File: package.json
Comment (MatissJanis) [inline package.json:41]: Would be really nice to finally get husky + lint-staged (or similar) working again. I see soo many commits that have `lint` step broken. Gets pretty annoying..

(but out of the scope of your PR; just thought it's worth mentioning)

File: packages/api/package.json
Comment (MatissJanis) [inline packages/api/package.json:20]: Do unit tests still pass? Last time I tried to do this for the electron upgrade some tests started mysteriously failing.

**Review by MatissJanis — COMMENTED** [review]
Great stuff!

Will approve straight after my branch (and the PR) gets merged to `master`.

File: packages/api/package.json
Comment (MatissJanis) [reply packages/api/package.json:20]: Here's my _very thorough_ investigation: https://github.com/actualbudget/actual/pull/462#discussion_r1070625353 😅

File: package.json
Comment (j-f1) [reply package.json:41]: FWIW my experience with those tools has been not great — they don’t pick up the path to `node` from GH Desktop in many cases and they slow down the commit process. IMO editor integrations (if people want those) + checks in CI are sufficient

File: packages/api/package.json
Comment (j-f1) [reply packages/api/package.json:20]: Will see once your PR is merged and the CI runs as expected

File: package.json
Comment (MatissJanis) [reply package.json:41]: It's a question of individual productivity versus maintainer productivity. I'll open a discussion up about what we do here sometime later. For now we can definitely remove husky as it's not used either way.

File: packages/api/package.json
Comment (j-f1) [reply packages/api/package.json]: I see what you mean. Fun, reverted that change

Comment (j-f1) [reply packages/api/package.json]: Did some testing and the test is fine with v7.4.6 but breaks with v7.5.0. Nothing jumps out in the diff.

Comment (MatissJanis) [reply packages/api/package.json]: Diff: https://github.com/WiseLibs/better-sqlite3/compare/v7.4.6...v7.5.0

My initial guess would be that the upgrade of sqlite to v3.37.2 is at fault here. Not sure how though.. and not sure if even the tests we currently have are correct. They might be incorrect in `master` branch and the upgrade is just exposing the problem.

Comment (j-f1) [reply packages/api/package.json]: Figured out the underlying issue: With SQLite 3.37.0, the transaction IDs fetched in the second call to `updateCategoryRules` in the test are `[ 'five', 'four', 'six', 'one', 'three', 'two' ]` but with 3.37.2 the IDs are `[ 'one', 'two', 'three', 'four', 'five', 'six' ]`. Reverting that change and will explore in a future PR… turns out the order rows are returned is implementation defined unless you specify `ORDER BY`.

Comment (MatissJanis) [reply packages/api/package.json]: Good investigation! I was trying to figure it out too just now, but didn't get as far as you :)

**Review by MatissJanis — APPROVED** [review]

File: packages/loot-core/package.json
Comment (MatissJanis) [inline packages/loot-core/package.json:53]: Did you intentionally downgrade `jest`?

Comment (j-f1) [reply packages/loot-core/package.json:53]: Yes, the latest `react-scripts` is on Jest 27 (and we pull in `jest-environment-jsdom` implicitly from that), so we need to use Jest 27 here too.

**Review by MatissJanis — APPROVED** [review]

**Pull request merged**
